src/pgen/cluster.cpp

```diff
@@ -405,4 +432,101 @@ void ProblemGenerator(MeshBlock *pmb, parthenon::ParameterInput *pin) {
   } // END if(hydro_pkg->Param<Fluid>("fluid") == Fluid::glmmhd)
 }
 
+void UserWorkBeforeOutput(MeshBlock *pmb, ParameterInput *pin) {
+  // get hydro
+  auto pkg = pmb->packages.Get("Hydro");
+  const Real gam = pin->GetReal("hydro", "gamma");
+  const Real gm1 = (gam - 1.0);
+
+  // get prim vars
+  auto &data = pmb->meshblock_data.Get();
+  auto const &prim = data->Get("prim").data;
+
+  // get derived fields
+  auto &log10_radius = data->Get("log10_cell_radius").data;
+  auto &entropy = data->Get("entropy").data;
+  auto &mach_sonic = data->Get("Mach_sonic").data;
+  auto &temperature = data->Get("temperature").data;
+
+  // for computing temperature from primitives
+  auto mbar_over_kb = pkg->Param<Real>("mbar_over_kb");
+
+  // fill derived vars (*including ghost cells*)
+  auto &coords = pmb->coords;
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+
+  pmb->par_for(
+      "FillDerivedHydro", kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        // get gas properties
+        const Real rho = prim(IDN, k, j, i);
+        const Real v1 = prim(IV1, k, j, i);
+        const Real v2 = prim(IV1, k, j, i);
+        const Real v3 = prim(IV1, k, j, i);
+        const Real P = prim(IPR, k, j, i);
+
+        // compute radius
+        const Real x = coords.Xc<1>(i);
+        const Real y = coords.Xc<2>(j);
+        const Real z = coords.Xc<3>(k);
+        const Real r2 = SQR(x) + SQR(y) + SQR(z);
+        log10_radius(0, k, j, i) = 0.5*std::log10(r2);
+
+        // compute entropy
+        const Real K = P / std::pow(rho, gam);
```

Comment:
Is this consistent with our other entropy definitions?

Reply:
Good point. I implemented this to be consistent with how I've defined it in the precipitator pgen, which is not necessarily consistent with how it's defined here.

Reply:
@pgrete I think I got it wrong from the notes I was looking at, thinking that the `5/3` came from a convention in the ACCEPT dataset verified in Mark's papers. After seeing this in Ben's PR, I found that this is a broader [Astrophysical Entropy](https://en.wikipedia.org/wiki/Entropy_(astrophysics)) so we should fix the entropy definition in the Hydrostatic Equilibrium PR if we want to use an adiabatic index that's not 5/3.

Reply:
For the cluster setup we might want to stick with the def you put in the doc (i.e, with respect to the electron number density) for easier comparison with the other result, don't we?
(This is independent of the the issue around using the adiabatic index #65 )